packages/patternfly-4/react-core/src/components/Tabs/Tabs.js

```diff
@@ -0,0 +1,221 @@
+      let selectedTab;
+      const childrenArr = Array.from(container.children);
+      childrenArr.forEach(child => {
+        const { className } = child;
+        if (className.search('pf-m-current') > 0) {
+          selectedTab = child;
+        }
+      });
+
+      const sideOutOfView = sideElementIsOutOfView(container, selectedTab);
+
+      this.setState({
+        showLeftScrollButton,
+        showRightScrollButton,
+        highlightLeftScrollButton: (sideOutOfView === SIDE.LEFT || sideOutOfView === SIDE.BOTH) && showLeftScrollButton,
+        highlightRightScrollButton:
+          (sideOutOfView === SIDE.RIGHT || sideOutOfView === SIDE.BOTH) && showRightScrollButton
+      });
+    }
+  };
+
+  scrollLeft = () => {
+    // find first Element that is fully in view on the left, then scroll to the element before it
+    if (this.tabList.current) {
+      const container = this.tabList.current;
+      const childrenArr = Array.from(container.children);
+      let firstElementInView;
+      let lastElementOutOfView;
+      let i;
+      for (i = 0; i < childrenArr.length && !firstElementInView; i++) {
+        if (isElementInView(container, childrenArr[i], false)) {
+          firstElementInView = childrenArr[i];
+          lastElementOutOfView = childrenArr[i - 1];
+        }
+      }
+      lastElementOutOfView && (container.scrollLeft -= lastElementOutOfView.scrollWidth);
+    }
+  };
+
+  scrollRight = () => {
+    // find last Element that is fully in view on the right, then scroll to the element after it
+    if (this.tabList.current) {
+      const container = this.tabList.current;
+      const childrenArr = Array.from(container.children);
+      let lastElementInView;
+      let firstElementOutOfView;
+      let i;
+      for (i = childrenArr.length - 1; i >= 0 && !lastElementInView; i--) {
+        if (isElementInView(container, childrenArr[i], false)) {
+          lastElementInView = childrenArr[i];
+          firstElementOutOfView = childrenArr[i + 1];
+        }
+      }
+      firstElementOutOfView && (container.scrollLeft += firstElementOutOfView.scrollWidth);
+    }
+  };
+
+  componentDidMount() {
+    window.addEventListener('resize', this.handleScrollButtons, false);
+
+    // call the handle resize function to check if scroll buttons should be shown
+    this.handleScrollButtons();
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('resize', this.handleScrollButtons, false);
+  }
+
+  render() {
+    const {
+      className,
+      children,
+      activeKey,
+      isFilled,
+      isSecondary,
+      leftScrollAriaLabel,
+      rightScrollAriaLabel,
+      ...props
+    } = this.props;
+    return (
+      <React.Fragment>
+        <div
+          {...props}
+          className={css(
+            styles.tabs,
+            isFilled && styles.modifiers.fill,
+            isSecondary && styles.modifiers.tabsSecondary,
+            this.state.showLeftScrollButton && styles.modifiers.start,
```

Comment:
can we destructure these state values like we do for the props?

Reply:
good idea.  thanks.